spec/tapioca/compilers/dsl/state_machines_spec.rb

```diff
@@ -0,0 +1,306 @@
+            end
+
+            event :disable do
+              transition active: :off
+            end
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        class Vehicle
+          include Vehicle::StateMachineInstanceHelperModule
+          extend Vehicle::StateMachineClassHelperModule
+        end
+
+        module Vehicle::StateMachineClassHelperModule
+          sig { params(event: T.any(String, Symbol)).returns(String) }
+          def human_alarm_state_event_name(event); end
+
+          sig { params(state: T.any(String, Symbol)).returns(String) }
+          def human_alarm_state_name(state); end
+        end
+      RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it(" generate RBI for a classe with state_machine, verify path, event and state helper methods ") do
+      content = <<~RUBY
+        class Vehicle
+          state_machine :alarm_state, initial: :active, namespace: :'alarm' do
+            state :active, :value => 1
+            state :off, :value => 0
+          end
+        end
+
+      RUBY
+      expected = <<~RUBY
+        module Vehicle::StateMachineInstanceHelperModule
+          sig { returns(T::Boolean) }
+          def alarm_active?; end
+
+          sig { returns(T::Boolean) }
+          def alarm_off?; end
+
+          sig { returns(Integer) }
+          def alarm_state; end
+
+          sig { params(value: Integer).returns(Integer) }
+          def alarm_state=(value); end
+
+          sig { params(state: T.any(String, Symbol)).returns(T::Boolean) }
+          def alarm_state?(state); end
+
+          sig { params(args: T.untyped).returns(T::Array[T.any(String, Symbol)]) }
+          def alarm_state_events(*args); end
+
+          sig { returns(T.any(String, Symbol)) }
+          def alarm_state_name; end
+
+          sig { params(args: T.untyped).returns(T::Array[::StateMachines::Transition]) }
+          def alarm_state_paths(*args); end
+
+          sig { params(args: T.untyped).returns(T::Array[::StateMachines::Transition]) }
+          def alarm_state_transitions(*args); end
+
+          sig { params(event: T.any(String, Symbol), args: T.untyped).returns(T::Boolean) }
+          def fire_alarm_state_event(event, *args); end
+
+          sig { returns(String) }
+          def human_alarm_state_name; end
+        end
+      RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it("generate RBI for a classe with state_machine, verify path helper method only ") do
+      content = <<~RUBY
+        class Vehicle
+          attr_accessor :seatbelt_on, :time_used, :auto_shop_busy
+
+          state_machine :state, initial: :parked do
+            before_transition parked: any - :parked, do: :put_on_seatbelt
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        sig { params(args: T.untyped).returns(T::Array[::StateMachines::Transition]) }
+          def state_paths(*args); end
```

Comment:
Can we fix the indentation here please?

Reply:
This was added by me so I can chip in.

Indentation is a result of HEREDOC and `include`. 

Generated string is:
```
⎵⎵sig { ...
⎵⎵def state_paths
```

Since in a squiggly HEREDOC the shortest indentation length is "subtracted" from every line our expected string with *nice* formatting becomes
```
sig { ...
def state_paths
```

This expected string fails the `include` check as it expects 2 whitespaces before `def`

I couldn't come up with another solution other than adding indentation

Reply:
@Ufuk, do you have any suggestion?